cpp/foxglove/include/foxglove/context.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string>
+
+struct foxglove_context;
+
+namespace foxglove {
+
+typedef foxglove_context ContextInner;
+
+class Context final {
+  friend class McapWriterOptions;
+  friend class Channel;
+  friend class WebSocketServerOptions;
+
+public:
+  Context();
+
+  static Context get_default();
```

Comment:
naming: I think we were starting to use camelCase for members (at least data members) in some other places, so might want to double-check for consistency (I don't really have a strong opinion of which way is right for C++, although I find mixing camel for types and snake for methods to be a little weird)

Reply:
That's a good question, I'd rather we stick to rust naming TitleCase for types and snake_case for methods and members. But that's just my preference. I'd like to see what the others say.